src/Lucene.Net/Util/UnmanagedStringArray.cs

```diff
@@ -215,6 +215,8 @@ public void Add(Span<char> str)
 
         public void AddEmpty()
         {
+            // since we are doing a binary search, we must keep the order of the terms
+            _strings[_index].Start = _index > 0 ? _strings[_index - 1].Start : null;
```

Comment:
For `_index == 0` we'll always have `null`. This means that in binary search this item will be always considered the lowest, won't it?. I have some doubts whether that is fine. We won't preserve the original order. Maybe for element at `0` position we need to _always_ add the real term value

CC @maciejaszyk

Reply:
Also current fix won't work for element at position `1` since at `_index - 1` will be `null`